userspace/libsinsp/procfs_utils.h

```diff
@@ -27,5 +27,43 @@ int get_userns_root_uid(std::istream& uid_map);
  */
 std::string get_systemd_cgroup(std::istream& cgroups);
 
-}
-}
+/**
+ * @brief Access container data through proc
+ */
+class ns_helper
+{
+public:
+	ns_helper(const std::string& host_root);
+	~ns_helper();
+
+	bool can_read_host_init_ns_mnt() const
+	{
+		return !m_cannot_read_host_init_ns_mnt;
+	}
+
+	const char* get_host_init_ns_mnt() const { return m_host_init_ns_mnt; }
+
+	//! Return true if not in the host init mount namespace
+	bool in_own_ns_mnt(int64_t pid) const;
+
+	std::string get_pid_root(int64_t pid) const
+	{
+		return m_host_root + "/proc/" + std::to_string(pid) + "/root";
+	}
+
+private:
+	const std::string& m_host_root;
+	char* m_host_init_ns_mnt{nullptr};
```

Comment:
Using a unique ptr?

Reply:
That's being used with C functions, I just felt it natural to keep it C-style :)